src/Plugins/ComplexCore/test/RawBinaryReaderTest.cpp

```diff
@@ -0,0 +1,446 @@
+#include "complex/Parameters/ArrayCreationParameter.hpp"
+#include "complex/Parameters/ChoicesParameter.hpp"
+#include "complex/Parameters/FileSystemPathParameter.hpp"
+#include "complex/Parameters/NumberParameter.hpp"
+#include "complex/Parameters/NumericTypeParameter.hpp"
+
+#include <filesystem>
+namespace fs = std::filesystem;
+
+#include "ComplexCore/Filters/RawBinaryReaderFilter.hpp"
+#include "complex/unit_test/complex_test_dirs.hpp"
+
+using namespace complex;
+
+const std::string k_TestOutput = (fs::path(complex::unit_test::k_BinaryDir.str()).append("RawBinaryReaderTest").append("Output.bin")).string();
+const DataPath k_CreatedArrayPath = DataPath(std::vector<std::string>{"Test_Array"});
+
+constexpr int32_t k_RbrNumComponentsError = -392;
+constexpr int32_t k_RbrWrongType = -393;
+constexpr int32_t k_RbrSkippedTooMuch = -395;
+
+namespace Detail
+{
+enum Endian
+{
+  Little = 0,
+  Big
+};
+} // namespace Detail
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+Arguments createFilterArguments(complex::NumericType scalarType, usize N, usize skipBytes)
+{
+  Arguments args;
+
+  args.insertOrAssign(RawBinaryReaderFilter::k_InputFile_Key, std::make_any<FileSystemPathParameter::ValueType>(fs::path(k_TestOutput)));
+  args.insertOrAssign(RawBinaryReaderFilter::k_ScalarType_Key, std::make_any<NumericType>(scalarType));
+  args.insertOrAssign(RawBinaryReaderFilter::k_NumberOfComponents_Key, std::make_any<uint64>(N));
+  args.insertOrAssign(RawBinaryReaderFilter::k_Endian_Key, std::make_any<ChoicesParameter::ValueType>(Detail::Little));
+  args.insertOrAssign(RawBinaryReaderFilter::k_SkipHeaderBytes_Key, std::make_any<uint64>(skipBytes));
+  args.insertOrAssign(RawBinaryReaderFilter::k_CreatedAttributeArrayPath_Key, k_CreatedArrayPath);
+
+  return args;
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+template <typename T>
+bool createAndWriteToFile(std::string filePath, T* dataArray, usize dataSize)
+{
+  // Create the output file to dump some data into
+  FILE* f = fopen(filePath.c_str(), "wb");
+
+  // Write the data to the file
+  usize numWritten = 0;
+  while(1)
+  {
+    numWritten += fwrite(dataArray, sizeof(T), dataSize, f);
+    if(numWritten == dataSize)
+    {
+      break;
+    }
+    dataArray = dataArray + numWritten;
+  }
+
+  // Close the file
+  fclose(f);
+
+  // Return successful
+  return true;
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+// Case1: This tests when skipHeaderBytes equals 0, and checks to see if the data read is the same as the data written.
+template <typename T, usize N>
+void testCase1_Execute(complex::NumericType scalarType)
```

Comment:
The changes for this test case are similar for all the others. Maybe some more common functions could be factored out.